test/functional/keyed_items.js

```diff
@@ -169,6 +169,23 @@ describe('rendering with keys', () => {
     expect(firstNode.parentNode).to.equal(null);
   });
 
+  it('should not throw error if rendering content on parent with outdated key map', () => {
+    function render(tag) {
+      elementVoid(tag, 'key');
+    }
+
+    patch(container, render, 'div');
+    const firstNode = container.childNodes[0];
+
+    container.removeChild(container.childNodes[0]);
+
+    expect(() => patch(container, render, 'span')).to.not.throw(Error);
+    const newNode = container.childNodes[0];
+    expect(newNode).not.to.equal(firstNode);
```

Comment:
Yes, I think you're right. So, just to make sure I understand, would you prefer if I remove all asserts from this test, just letting it fail if an error gets thrown?